sdk/ml/azure-ai-ml/azure/ai/ml/_restclient/v2024_04_01_preview/__init__.py

```diff
@@ -0,0 +1,18 @@
+# coding=utf-8
```

Comment:
I was expecting to see `MLClient` to start using this API version but I don't see it in this PR.

Reply:
The actual API calls haven't changed (or at least not the ones I'm concerned with). The new rest client was needed in order to properly reference new conenction categories and rest connection classes.

Reply:
I'm not fully getting this. If we didn't switch to new service client, how did we start to get new rest classes?